lib/puppet/catalog-diff/compilecatalog.rb

```diff
@@ -122,6 +125,46 @@ def render_pson(catalog)
       pson
     end
 
+    def clean_sensitive_parameters(catalog)
+      catalog['resources'].map! do |resource|
+        if resource.key? 'sensitive_parameters'
+          resource['sensitive_parameters'].each do |p|
+            hash = Digest::SHA256.hexdigest Marshal.dump(resource['parameters'][p])
+            resource['parameters'][p] = 'Sensitive [hash ' + hash + ']'
+          end
+          resource.tap { |hs| hs.delete('sensitive_parameters') }
```

Comment:
What is `#tap` for?

Reply:
I'll own up to taking this from stackoverflow! https://stackoverflow.com/a/9027559

Reply:
But why not modify the original hash?

Reply:
ah, ok.  Doh!  I should have looked more carefully.  The original poster was after a one-liner.  I had just forgotten whether you set a key to `nil` to delete it, or if there was a method to use.

Reply:
Actually, do we really need to remove the `sensitive_parameters` field, if it's only the list of keys?

Reply:
yes, I think so. Otherwise our useful hashes get replaced by standard puppet `redacted` stuff sometime later.  IIRC I think during https://github.com/camptocamp/puppet-catalog-diff/blob/b5d0f72d37a58948f45c920caaa39a4a9c90cf44/lib/puppet/catalog-diff/differ.rb#L28? (But would have to double check to be sure if it was after this line or somewhere else)